iron-location.html

```diff
@@ -181,7 +181,11 @@
             this.path).replace(/\#/g, '%23').replace(/\?/g, '%3F');
         var partiallyEncodedQuery = '';
         if (this.query) {
-          partiallyEncodedQuery = '?' + this.query.replace(/\#/g, '%23');
+          // decoding here needed to prevent encoding the % sign. decodeURIComponent
+          // because it will not decode the hash otherwise. encodeURI and not
+          // component because it will encode ? to %3F otherwise.
```

Comment:
It may be accepted (like spaces), but I don't think that makes the URL valid. I'll defer to another reviewer to since I'm not sure what we should do in this case.